clean/ca/grass_valley_pd.py

```diff
@@ -0,0 +1,142 @@
+        for link in links:
+            link_href = link.get("href", None)
+            print(link_href)
+            link_href = f"{self.base_url}{link_href}"
+            case_id = link.string
+            if link_href:
+                title = link.string
+                child_name = f"{link_href.split('/')[-1]}.html"
+                child_filename = f"{self.agency_slug}/{child_name}"
+                self.cache.download(child_filename, link_href, force=True)
+                html = self.cache.read(child_filename)
+                soup = BeautifulSoup(html, "html.parser")
+                content_areas = soup.find("section", class_="page-content")
+                child_links = content_areas.find_all("a")
+                video_counter = 1
+                for child_link in child_links:
+                    link = child_link.get("href", None)
+                    if "file-attachments" in link:
+                        title = child_link.string
+                        name = link.split("/")[-1].split("?")[0]
+                        payload = {
+                            "asset_url": f"{self.base_url}{link}",
+                            "case_id": case_id,
+                            "name": name,
+                            "title": title,
+                            "parent_page": str(child_filename),
+                        }
+                        metadata.append(payload)
+                    if "sharepoint" in link:
+                        download_link = self._get_sharepoint_link(link)
+                        if download_link:
+                            name = download_link.split("/")[-1]
+                            title = child_link.string.strip()
+                            payload = {
+                                "asset_url": download_link,
+                                "case_id": case_id,
+                                "name": name,
+                                "title": title,
+                                "parent_page": str(child_filename),
+                            }
+                            metadata.append(payload)
+                    if "vimeo" in link:
+                        link = link.split("?")[0]
+                        name = link.split("/")[-1]
+                        title = f"{case_id}-{video_counter}"
+                        video_counter += 1
+                        payload = {
+                            "asset_url": link,
+                            "case_id": case_id,
+                            "name": name,
+                            "title": title,
+                            "parent_page": str(child_filename),
+                        }
+                        metadata.append(payload)
+
+            time.sleep(throttle)
+
+        outfile = self.data_dir.joinpath(f"{self.agency_slug}.json")
+        self.cache.write_json(outfile, metadata)
+        return outfile
+
+    def _get_sharepoint_link(self, url):
```

Comment:
Suggestion: move this into a util that accepts an additional argument, `download_url`